packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/TransactionModal.tsx

```diff
@@ -0,0 +1,227 @@
+"use client";
+import * as DialogPrimitive from "@radix-ui/react-dialog";
+import { ArrowTopRightIcon } from "@radix-ui/react-icons";
+import cn from "classnames";
+import { PropsWithChildren, useContext, useState } from "react";
+
+import { Button } from "#/components";
+import { Toast } from "#/components/Toast";
+import { PoolMetadataContext } from "#/contexts/PoolMetadataContext";
+import { TransactionStatus, useTransaction } from "#/hooks/useTransaction";
+
+type Stage = {
+  pinColor: string;
+  confirmColor: string;
+  writeColor: string;
```

Comment:
not clear what these things refer to just by looking at the names here 👁️

Reply:
I'll try to be more clear here